packages/compass-workspace/src/components/tab.tsx

```diff
@@ -0,0 +1,314 @@
+  color: uiColors.gray.light2,
+});
+
+type TabProps = {
+  activeSubTabName: string;
+  darkMode?: boolean;
+  isFocused: boolean;
+  isSelected: boolean;
+  onSelect: () => void;
+  onClose: () => void;
+  tabId: string;
+  namespace: string;
+  type: TabType;
+  isTabListFocused: boolean;
+};
+
+function UnthemedTab({
+  darkMode,
+  activeSubTabName,
+  isFocused,
+  isSelected,
+  isTabListFocused,
+  onSelect,
+  onClose,
+  tabId,
+  type,
+  namespace,
+}: TabProps) {
+  const defaultActionProps = useDefaultAction(onSelect);
+
+  const [hoverProps, isHovered] = useHoverState();
+
+  const tabProps = mergeProps<HTMLDivElement>(hoverProps, defaultActionProps);
+
+  const tabIcon = useMemo(() => {
+    return type === 'timeseries'
+      ? 'TimeSeries'
+      : type === 'view'
+      ? 'Visibility'
+      : 'Folder';
+  }, [type]);
+
+  return (
+    <div
+      className={cx(
+        tabStyles,
+        darkMode ? tabDarkThemeStyles : tabLightThemeStyles,
+        {
+          [selectedTabStyles]: isSelected,
+          [focusedTabStyles]: isFocused,
+        }
+      )}
+      aria-selected={isSelected}
+      role="tab"
+      // The tab navigation is handled by the lab list.
+      tabIndex={-1}
+      aria-controls={tabId}
+      title={`${namespace} - ${activeSubTabName}`}
+      {...tabProps}
+    >
+      <Icon
+        className={cx(tabIconStyles, {
+          [darkMode
+            ? tabIconSelectedDarkThemeStyles
+            : tabIconSelectedLightThemeStyles]: isSelected,
+          [tabIconFocusedStyles]: isFocused,
+        })}
+        role="presentation"
+        glyph={tabIcon}
+        size="small"
+      />
+      <div className={tabTitleContainerStyles}>
+        <div
+          className={cx(
+            tabTitleStyles,
+            darkMode ? tabTitleDarkThemeStyles : tabTitleLightThemeStyles,
+            {
+              [darkMode
+                ? tabTitleSelectedDarkThemeStyles
+                : tabTitleSelectedLightThemeStyles]: isSelected,
+              [tabTitleFocusedStyles]: isFocused,
+            }
+          )}
+        >
+          {activeSubTabName}
+        </div>
+        <Body
+          className={cx(
+            tabSubtitleStyles,
+            darkMode ? tabSubtitleDarkThemeStyles : tabSubtitleLightThemeStyles,
+            {
+              [darkMode
+                ? tabSubtitleSelectedDarkThemeStyles
+                : tabSubtitleSelectedLightThemeStyles]: isSelected,
+            }
+          )}
+        >
+          {namespace}
+        </Body>
+      </div>
+
+      <IconButton
+        className={
+          (isSelected && isTabListFocused) || isFocused || isHovered
+            ? undefined
+            : hiddenStyles
+        }
+        onClick={(e) => {
+          e.stopPropagation();
+          onClose();
+        }}
+        aria-label="Close Tab"
+      >
+        <Icon glyph="X" role="presentation" />
+      </IconButton>
+      <div
+        role="presentation"
```

Comment:
It's not bad to have this role here, but you don't really need it for nodes without content